test/test_torch.py

```diff
@@ -938,6 +939,240 @@ def test_add(self):
 
         # [res] torch.add([res,] tensor1, value, tensor2)
 
+    def mb_rand(self, *dims):
+        dims = [dim for dim in dims if dim != ()]
+        xs = [torch.rand(1, *(random.randint(1, size) if b else size for b, size in dims[1:])) for i in range(dims[0])]
+        xb = MaskedBatch.fromlist(xs, tuple(b for b, d in dims[1:]))
+        return xs, (xb.data, xb.mask, xb.dims)
+
+    def mb_assert_allclose(self, xs, ybs):
+        if isinstance(ybs, torch.autograd.Variable):
+            self.assertAlmostEqual(xs.data, ybs.data, places=3)
+        elif isinstance(ybs, MaskedBatch):
+            self.mb_assert_allclose(xs, ybs.examples())
+        else:
+            if isinstance(ybs, (list, tuple)):
+                for j, yb in enumerate(ybs):
+                    for i, y in enumerate(yb.examples()):
+                        self.mb_assert_allclose(xs[i][j], y)
+            else:
+                for x, yb in zip(xs, ybs):
+                    self.mb_assert_allclose(x, yb)
```

Comment:
It would be better to override `assertEqual` and add a special case to unpack `MaskedBatch`, but generally leave comparing the tensors to the superclass implementation. It already does allclose on tensors by default.